src/components/main.tsx

```diff
@@ -85,11 +119,11 @@ const Main = () => {
                         )
                         :
                         <Button
-                            onClick={() => toggleShowQR(!showQR)}
+                            onClick={submitAddress}
```

Comment:
We actually have two submit buttons, this one, and the one above for mobile devices which does:
`onClick={() => openAppOrAppStore()}` 
so you will need to change it for that one as well. You can handle it in the same function:
```javascript
    const submitAddress = () => {
      if(validAddress && !isMobile) toggleShowQR(true);
      else if (validAddress && isMobile) openAppOrAppStore();
      else setInvalidError(true)
    }
```